### ISSUE TITLE
Brief 2.3.0 settings do not keep unless page is refreshed before closing

### ISSUE THREAD

**Issue by sleek22** [ancestorak/brief#243]
This is a minor problem. But it might help someone who might be having the same issue.

I am on Windows 10, pcxFirefox 64-bit. I am using the Brief icon to open the Brief page.

I have a sidebar that is too wide and a folder that is expanded which I want closed. Changing the settings has no effect when I open Brief again. Only by refreshing the Brief page after changes do the settings keep.

As I brought this up in another post, tanriol kindly pointed out that Brief settings are now stored in extensions.brief.pagePersist and that he could not reproduce the problem. That allowed me to do some more troubleshooting.

My extensions.brief.pagePersist looks like this and does not change with changes to Brief:

{"startView":"all-items-folder","closedFolders":"_","tagList":{"width":""},"sidebar":{"width":"","hidden":false},"view":{"filter":"unread","mode":"full"}}

Only when I refresh does it change to this:

{"startView":"all-items-folder","closedFolders":"_99648_","tagList":{"width":""},"sidebar":{"width":"327px","hidden":false},"view":{"filter":"unread","mode":"full"}}

As you can see, the width is now set and I confirmed that the folder in question is the id of the folder that was being expanded, so it does now keep the settings, even across Firefox restarts.

If anyone wants to test this, open up Brief and about:config pointed to extensions.brief.pagePersist side by side and then change the sidebar width. In my case, the about:config settings do not change no matter what I do. Only when I refresh the Brief page, then they change.

So basically a page refresh, at least in my case, keeps the settings.

**Comment by sleek22** [comment]
Sorry, I should have also mentioned that I'm on Firefox 52.0.2 in case that makes a difference.

**Comment by sleek22** [comment]
Update...

I have only two tabs open----- about:config on the left, Brief on the right.

I change the sidebar width, sit on the about:config page after changing the Brief sidebar width, and use a FireGestures stroke to close all tabs to right. I do this ten times. Every time Brief closes, there is no change to Brief settings.

I change the sidebar width, sit on the about:config page after changing the Brief sidebar width, and right-click on the about:config tab context menu to close all tabs to right. I do this ten times. Every time Brief closes, the Brief settings change.

In both cases, I am sitting on the about:config page when closing Brief. It seems that even though in both cases the Brief tab is closing, it makes a difference whether the tab is closed by context menu or by using FireGestures to close it.

I did not mention in my first comment that I close tabs with FireGestures strokes but I can see now that it makes a difference.

I've also tested by closing Brief by clicking on its tab and clicking on Close Tab on its context menu. In that case, it also keeps the settings.

So something in the way tabs are closed...
